node/node.go

```diff
@@ -674,9 +691,13 @@ func (node *Node) setupForShardLeader() {
 	node.serviceManager.RegisterService(service_manager.BlockProposal, blockproposal.NewService(node.Consensus.ReadySignal, node.WaitForConsensusReady))
 	// Register client support service.
 	node.serviceManager.RegisterService(service_manager.ClientSupport, clientsupport.NewService(node.blockchain.State, node.CallFaucetContract, node.SelfPeer.IP, node.SelfPeer.Port))
+	// Register peer discovery service.
+	node.serviceManager.RegisterService(service_manager.PeerDiscovery, discovery.New(node.host, fmt.Sprintf("%v", node.Consensus.ShardID)))
```

Comment:
as discussed you said peerdiscovery does 2 things: 1/ for new node, it talks to bootnode to obtain info of some beacon peers
2/ maintain the discovery peers

i believe we need 2/ for every node.
1/ is just for new node or anyone who does not know how to join the blockchain network.

with that i'm thinking we can break down discovery into 2 different services.
what do you think?

Reply:
it is okay to split.  So I will use networkinfo service to connect to bootnode and discovery service to connect to peers.